components/script/dom/htmliframeelement.rs

```diff
@@ -112,7 +112,9 @@ impl HTMLIFrameElement {
         (old_pipeline_id, new_pipeline_id)
     }
 
-    pub fn navigate_or_reload_child_browsing_context(&self, load_data: Option<LoadData>, replace: bool) {
+    pub fn navigate_or_reload_child_browsing_context(&self, load_data: Option<LoadData>,
+                                                     nav_type: NavigationType,
```

Comment:
Rather than add the `nav_type` parameter, would it be easier just to have this function do navigation and reloading, and put the code for the initial `about:blank` document into `create_nested_browsing_context`? It looks like most of the code for the two cases is distinct.